### PR TITLE
fix(ref-imp): fix cas uri validation

### PR DESCRIPTION
**Pull request by isaacJChen**

### CODE REVIEW

**Comment by isaacJChen** [comment]
#927

File: lib/core/versions/latest/InputValidator.ts [hunk 1]
```diff
@@ -51,12 +51,10 @@ export default class InputValidator {
       );
     }
 
-    const casFileUriAsHashBuffer = Encoder.decodeAsBuffer(casFileUri);
-    const hashAlgorithmInMultihashCode = ProtocolParameters.hashAlgorithmInMultihashCode;
-    if (!Multihash.isComputedUsingHashAlgorithm(casFileUriAsHashBuffer, hashAlgorithmInMultihashCode)) {
+    if (casFileUri.length > ProtocolParameters.maxCasUriLength) {
       throw new SidetreeError(
-        ErrorCode.InputValidatorCasFileUriUnsupported,
```
Comment (thehenrytsai) [inline lib/core/versions/latest/InputValidator.ts:58]: Sanity: Remove in `ErrorCode` if `InputValidatorCasFileUriUnsupported` no longer referenced.

File: lib/core/versions/latest/InputValidator.ts [hunk 2]
```diff
@@ -51,12 +51,10 @@ export default class InputValidator {
...
-        `Input ${inputContextForErrorLogging} CAS file URI '${casFileUri}' is not computed using hash algorithm of code ${hashAlgorithmInMultihashCode}.`
+        ErrorCode.InputValidatorCasFileUriExceedsMaxLength,
+        `Input ${inputContextForErrorLogging} CAS file URI '${casFileUri}' length needs to be less than ${ProtocolParameters.maxCasUriLength}`
```
Comment (thehenrytsai) [inline lib/core/versions/latest/InputValidator.ts:57]:
```suggestion
        `Input ${inputContextForErrorLogging} CAS file URI '${casFileUri}' length cannot exceed ${ProtocolParameters.maxCasUriLength}`
```

File: tests/core/CoreIndexFile.spec.ts [hunk 3]
```diff
@@ -41,19 +41,19 @@ describe('CoreIndexFile', async () => {
...
+      const coreIndexFileUri = 'EiB4ypIXxG9aFhXv2YC8I2tQvLEBbQAsNzHmph17vMfVYA';
       const coreProofFileUri = 'EiBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBBB'; // Should not be allowed with no recovers and deactivates.
 
       // Create operation.
       const createOperationData = await OperationGenerator.generateCreateOperation();
       const createOperation = createOperationData.createOperation;
 
       const coreIndexFileBuffer =
-        await CoreIndexFile.createBuffer(undefined, provisionalIndexFileUri, coreProofFileUri, [createOperation], [], []);
+        await CoreIndexFile.createBuffer(undefined, coreIndexFileUri, coreProofFileUri, [createOperation], [], []);
```
Comment (thehenrytsai) [inline tests/core/CoreIndexFile.spec.ts]: Sanity: I think you are incorrectly changing these, `provisonalIndexFileUri` is correct.

Comment (isaacJChen) [reply tests/core/CoreIndexFile.spec.ts]: Reverted, it was a mistake.

**Review by thehenrytsai — DISMISSED** [review]

**Review by sandrask — APPROVED** [review]

**Review by thehenrytsai — APPROVED** [review]

**Pull request merged**